gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,634 @@
+    aou_ht = meta_ht.filter(meta_ht.project == "aou").select()
+
+    aou_ht = aou_ht.annotate(hard_filters=aou_hard_filters_ht[aou_ht.key])
+
+    aou_ht = aou_ht.annotate(
+        hard_filters=hl.if_else(
+            hl.is_defined(aou_ht.hard_filters) & (hl.len(aou_ht.hard_filters) > 0),
+            {"sample_qc_metrics"},
+            hl.empty_set(hl.tstr),
+        )
+    )
+
+    # Add sample exclusion to hard filters if sample is in exclusion list.
+    aou_ht = aou_ht.annotate(
+        hard_filters=hl.if_else(
+            samples_to_exclude.contains(aou_ht.s),
+            aou_ht.hard_filters.union({"sample_exclusion"}),
+            aou_ht.hard_filters,
+        )
+    )
+
+    # Add AoU hard filter to the meta Table.
+    meta_ht = meta_ht.annotate(
+        hard_filters=hl.or_else(
+            hl.if_else(
+                meta_ht.project == "gnomad",
+                meta_ht.hard_filters,
+                aou_ht[meta_ht.key].hard_filters,
+            ),
+            hl.empty_set(hl.tstr),
+        )
+    )
+
+    meta_ht = meta_ht.annotate(
+        hard_filtered=hl.or_else(hl.len(meta_ht.hard_filters) > 0, False)
+    )
+
+    return meta_ht
+
+
+def annotate_genetic_ancestry(
+    meta_ht: hl.Table, v4_meta_ht: hl.Table, aou_gen_anc_ht: hl.Table
+) -> hl.Table:
+    """
+    Annotate metadata Table with genetic ancestry information.
+
+    .. note::
+
+        - For 'gnomad' project, uses v4 genetic ancestry inference.
+        - For 'aou' project, uses AoU projected genetic ancestry.
+
+    :param meta_ht: Table with metadata.
+    :param v4_meta_ht: Table with v4 genetic ancestry inference.
+    :param aou_gen_anc_ht: Table with AoU projected genetic ancestry.
+    :return: Table with genetic ancestry inference annotated.
+    """
+
+    # Prefix collisions for``.
+    v4_meta_ht = add_project_prefix_to_sample_collisions(
+        t=v4_meta_ht,
+        sample_collisions=sample_id_collisions.ht(),
+        project="gnomad",
+    )
+
+    # Nest all row fields (except key) under gen`etic_ancestry_inference for AoU data.
+    fields_to_nest = [f for f in aou_gen_anc_ht.row if f != "s"]
+    aou_gen_anc_ht = aou_gen_anc_ht.transmute(
+        genetic_ancestry_inference=hl.struct(
+            **{f: aou_gen_anc_ht[f] for f in fields_to_nest}
+        )
+    )
+
+    # Drop unnecessary fields from v4 meta that are not in the AoU metadata.
+    v4_meta_ht = v4_meta_ht.transmute(
+        genetic_ancestry_inference=v4_meta_ht.population_inference.drop(
+            "prob_oth", "training_pop_all", "training_pop"
+        )
+    )
+
+    # Rename 'pop' to 'gen_anc' while maintainf order of annotations.
+    v4_meta_ht = v4_meta_ht.annotate(
+        genetic_ancestry_inference=hl.struct(
+            gen_anc=v4_meta_ht.genetic_ancestry_inference.pop,
+            **{
+                k: v
+                for k, v in v4_meta_ht.genetic_ancestry_inference.items()
+                if k != "pop"
```

Comment:
```suggestion
                if k not in ["pop", "prob_oth", "training_pop_all", "training_pop"]
```
If you remove the piece above removing the fields and do it here.